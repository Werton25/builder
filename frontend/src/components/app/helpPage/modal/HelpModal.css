.ReactModal__Overlay {
    z-index: 10;
}

.helpModal {
    margin-top: 100px;
    width: 50%;
}
