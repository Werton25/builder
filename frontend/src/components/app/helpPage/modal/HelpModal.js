import React from 'react';
import PropTypes from 'prop-types';
import {isHelpOpen} from "../../../../selectors/modal";
import {toggleModal} from "../../../../actions/modal";
import {connect} from "react-redux";
import HelpContent from "../HelpContent";
import Modal from 'react-modal';
import './HelpModal.css';

export class BaseHelpModal extends React.Component {
  onToggleModal = () => {
    this.props.toggleModal('help');
  };

  render() {
    return (
      <Modal isOpen={this.props.isOpen}
             onRequestClose={this.onToggleModal}
             contentLabel="Горячие клавиши редактора запросов"
             className="modal-content center-block helpModal">
        <div className="modal-header">
          <button type="button" className="close" onClick={this.onToggleModal}>
            <span aria-hidden={true}>&times;</span>
          </button>
          <div className="modal-title text-center">Горячие клавиши редактора запросов</div>
        </div>
        <div className="modal-body">
          <HelpContent/>
        </div>
      </Modal>
    )
  }
}

BaseHelpModal.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  toggleModal: PropTypes.func.isRequired
};

const mapStateToProps = (state) => ({
  isOpen: isHelpOpen(state)
});

const mapDispatchToProps = {
  toggleModal
};

export default connect(mapStateToProps, mapDispatchToProps)(BaseHelpModal);
