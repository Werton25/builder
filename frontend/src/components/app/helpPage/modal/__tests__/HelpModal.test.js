import React from "react";
import renderer from "react-test-renderer";
import {BaseHelpModal} from "../HelpModal";

describe('HelpModal tests', () => {
  it('HelpModal renders correctly', () => {
    const tree = renderer
      .create(
        <BaseHelpModal isOpen={false} // https://github.com/reactjs/react-modal/issues/553
                       toggleModal={jest.fn()}/>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
