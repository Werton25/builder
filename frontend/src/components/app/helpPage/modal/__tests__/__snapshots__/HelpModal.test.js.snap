// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpModal tests HelpModal renders correctly 1`] = `null`;
